plugins/modules/aci_vrf_leak_internal_subnet.py

```diff
@@ -0,0 +1,408 @@
+
+# Copyright: (c) 2023, <> (@abmughal)
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "certified"}
+
+DOCUMENTATION = r"""
+---
+module: aci_vrf_leak_internal_subnet
+short_description: Manage contexts or VRFs (fv:leakInternalSubnet)
+description:
+- Manage leaking subnets under VRF.
+options:
+  tenant:
+    description:
+    - The name of the Tenant the VRF should belong to.
+    type: str
+    aliases: [ tenant_name ]
+  vrf:
+    description:
+    - The name of the VRF.
+    type: str
+    aliases: [ context, name, vrf_name ]
+  description:
+    description:
+    - The description for the VRF.
+    type: str
+    aliases: [ descr ]
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+  name_alias:
+    description:
+    - The alias for the current object. This relates to the nameAlias field in ACI.
+    type: str
+  match_type:
+    description:
+    - Configures match type for contracts under vzAny.
+    type: str
+    choices: [ all, at_least_one, at_most_one, none]
+  scope:
+    description:
+    - Scope of the object.
+    type: str
+    choices: [ public, private, shared ]
+    default: private
+  leak_internal_subnet:
+    description:
+    - The subnets being leaked to.
+    type: list
+    elements: dict
+    suboptions:
+      tenant:
+        description:
+        - Name of the tenant.
+        type: str
+        aliases: [ tenant_name ]
+      vrf:
+        description:
+        - Name of the VRF.
+        type: str
+        aliases: [ vrf_name ]
+  ip:
+    description:
+    - The IP address.
+    type: str
+extends_documentation_fragment:
+- cisco.aci.aci
+- cisco.aci.annotation
+- cisco.aci.owner
+
+notes:
+- The C(tenant) and C(vrf) used must exist before using this module in your playbook.
+  The M(cisco.aci.aci_tenant) and M(cisco.aci.aci_vrf) modules can be used for this.
+seealso:
+- module: cisco.aci.aci_tenant
+- module: cisco.aci.aci_vrf
+- name: APIC Management Information Model reference
+  description: More information about the internal APIC class B(leak:InternalSubnet).
+  link: https://developer.cisco.com/docs/apic-mim-ref/
+author:
+- Abraham Mughal (@abmughal)
+"""
+
+EXAMPLES = r"""
+- name: Create leak internal subnet
+  cisco.aci.aci_vrf_leak_internal_subnet:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    vrf: vrf_lab
+    tenant: lab_tenant
+    descr: Lab VRF
+    state: present
+    leak_internal_subnet:
+      - vrf: "test"
+        tenant: "lab_tenant"
+    description: Ansible Test
+    ip: 1.1.1.2
+  delegate_to: localhost
+
+- name: Remove a subnet from leaking
```

Comment:
1. What's the difference between this one and the example below? 
2. Shouldn't leak_internal_subnet have vrf and tenant in a list format like the above example?